### PR TITLE
Adds Template Tasks

### LINKED ISSUE #472: Redefining a running=true tick results in running=false with no error
* Kapacitor 0.12.0 / Linux amd64
* When redefining a running=false tick with "kapacitor define", the "executing" state goes to "true" (this is ok)
* When redefining a running=true tick with "kapacitor define", the "executing" state goes to "false" (this is the problem)
* The error message visible in "kapacitor show" output is always empty when this happens
* Workaround: I always run "kapacitor reload <tick>" after redefining a tick

Example session:

```
-bash-4.3# kapacitor define -name foo -tick foo.tick -dbrp "db86feee4290d74826ad9f52f53a383264"."default" -type stream; kapacitor show foo | head -5
Name: foo
Error:
Type: stream
Enabled: true
Executing: true

-bash-4.3# kapacitor define -name foo -tick foo.tick -dbrp "db86feee4290d74826ad9f52f53a383264"."default" -type stream; kapacitor show foo | head -5
Name: foo
Error:
Type: stream
Enabled: true
Executing: false

-bash-4.3# kapacitor define -name foo -tick foo.tick -dbrp "db86feee4290d74826ad9f52f53a383264"."default" -type stream; kapacitor show foo | head -5
Name: foo
Error:
Type: stream
Enabled: true
Executing: true
```

### LINKED ISSUE #483: Task is `Executing: false` without Error
I have a batch task the periodically fails shortly after starting in 12.0. It looks like if it successfully starts, it continues to run. Unfortunately, show task doesn't display an error and neither do the logs. I can trigger it inconsistently with reloading: `kapacitor reload disk_alert`

Reload logs when it fails (if it succeeds, the second `stopped` doesn't show):
```
[task_master] 2016/04/21 19:39:04 I! Stopped task: free_disk_alert
[httpd] 2016/04/21 19:39:04 I! 127.0.0.1 - - [21/Apr/2016:19:39:04 +0000] POST /disable?name=free_disk_alert HTTP/1.1 200 23 - Go-http-client/1.1 b48fd89c-07f8-11e6-812d-000000000000 13.723396ms
[task_master] 2016/04/21 19:39:04 I! Started task: free_disk_alert
[httpd] 2016/04/21 19:39:04 I! 127.0.0.1 - - [21/Apr/2016:19:39:04 +0000] POST /enable?name=free_disk_alert HTTP/1.1 200 23 - Go-http-client/1.1 b49211dc-07f8-11e6-812e-000000000000 9.388933ms
[task_master] 2016/04/21 19:39:04 I! Stopped task: free_disk_alert
[httpd] 2016/04/21 19:39:08 I! 127.0.0.1 - - [21/Apr/2016:19:39:08 +0000] GET /task?name=free_disk_alert HTTP/1.1 200 595 - Go-http-client/1.1 b73b9bc4-07f8-11e6-812f-000000000000 2.963007ms
```

Unsure what blank error means, but I'd guess the failure is coming from: https://github.com/influxdata/kapacitor/blob/00e6f4de33aaa9f6eba248d93e600c72a5591b06/task_master.go#L367

Show results:
```
Name: free_disk_alert
Error: 
Type: batch
Enabled: true
Executing: false
Databases Retention Policies: ["firehose"."default"]
TICKscript:
var msg = '{{ .ID }} is {{ .Level }}. {{  index .Tags "job"  }}/{{  index .Tags "index"  }} has {{ index .Fields "remaining_disk" }} GB free'

var data = batch
    |query('''
     	SELECT ( mean("remaining_disk") / 1024 ) AS remaining_disk
     	FROM "firehose"."default"./.*\.runner/
     ''')
        .period(20s)
        .every(10s)
        .groupBy('job', 'index')

data
    |alert()
        .id('Free Disk')
        .message(msg)
        .stateChangesOnly()
        .warn(lambda: int("remaining_disk") < 50)
        .Slack()

data
    |alert()
        .id('Free Disk')
        .message('QA {{  index .Tags "job"  }}/{{  index .Tags "index"  }} Free Disk is {{ .Level }}')
        .details(msg)
        .stateChangesOnly()
        .crit(lambda: int("remaining_disk") < 40)
        .email('someone@somewhere.com')


DOT:
digraph free_disk_alert {
batch1 -> alert2;
batch1 -> alert3;
}
```

### LINKED ISSUE #552: [feature request] Be able to reference TICK variables in alert message
Currently it is not possible to reference TICK script variables in alert messages. For example:

```
var WARN_THRESHOLD = 2
...
    |alert()
        .message('Broke threshold of {{ $WARN_THRESHOLD }}')
...
```

This can be useful for conveying message details that include the value that triggered the alert as well as what threshold/value the alert is firing from.

### PR DESCRIPTION
**Pull request by nathanielc**
With this change you can now define task templates. A simple example:

```go
// Which measurement to consume
var measurement string
// Optional where filter
var where_filter = lambda: TRUE
// Optional list of group by dimensions
var groups  = [*]
// Which field to process
var field string
// Warning criteria, has access to 'mean' field
var warn lambda
// Critical criteria, has access to 'mean' field
var crit lambda
// How much data to window
var window  = 5m
// The slack channel for alerts
var slack_channel = '#alerts'

stream
    |from()
        .measurement(measurement)
        .where(where_filter)
        .groupBy(groups)
    |window()
        .period(window)
        .every(window)
    |mean(field)
    |alert()
         .warn(warn)
         .crit(crit)
         .slack()
         .channel(slack_channel)
```

Then you can define the template like so:

```
kapacitor define-template generic_mean_alert -tick path/to/above/script.tick -type stream
```

Next define a task that uses the template:

```
kapacitor define cpu_alert -template-id generic_mean_alert -vars cpu_vars.json -dbrp telegraf.default
```

Where `cpu_vars.json` would like like this:

```json
{
    "measurement": {"type" : "string", "value" : "cpu" },
    "where_filter": {"type": "lambda", "value": "\"cpu\" == 'cpu-total'"},
    "groups": {"type": "list", "value": [{"type":"string", "value":"host"},{"type":"string", "value":"dc"}]},
    "field": {"type" : "string", "value" : "usage_idle" },
    "warn": {"type" : "lambda", "value" : " \"mean\" < 30.0" },
    "crit": {"type" : "lambda", "value" : " \"mean\" < 10.0" },
    "window": {"type" : "duration", "value" : "1m" },
    "slack_channel": {"type" : "string", "value" : "#alerts_testing" }
}
```

This fixes #552 via string concat with the improved TICKscript var evaluation.

This also fixes #472 and fixes #483 since the bug for those was causing new template tests to fail.

TODO
- [x] CHANGELOG
- [x] Created issue to track documenting new functionality
- [x] Specify vars on CLI
- [x] Parse string based durations

### CODE REVIEW

File: cmd/kapacitor/main.go [hunk 1]
```diff
@@ -1059,31 +1185,131 @@ func doShow(args []string) error {
...
-	fmt.Println("LastEnabled:", ti.LastEnabled.Format(time.RFC822))
-	fmt.Println("Databases Retention Policies:", ti.DBRPs)
-	fmt.Printf("TICKscript:\n%s\n\n", ti.TICKscript)
-	fmt.Printf("DOT:\n%s\n", ti.Dot)
+	fmt.Println("ID:", t.ID)
+	fmt.Println("Error:", t.Error)
+	fmt.Println("Template:", t.TemplateID)
+	fmt.Println("Type:", t.Type)
+	fmt.Println("Status:", t.Status)
+	fmt.Println("Executng:", t.Executing)
```
Comment (jsotelo) [inline cmd/kapacitor/main.go]: Typo here, should be "Executing"

File: cmd/kapacitor/main.go [hunk 2]
```diff
@@ -1059,31 +1185,131 @@ func doShow(args []string) error {
...
+	fmt.Println("Created:", t.Created.Format(time.RFC822))
+	fmt.Println("Modified:", t.Modified.Format(time.RFC822))
+	fmt.Println("LastEnabled:", t.LastEnabled.Format(time.RFC822))
+	fmt.Println("Databases Retenton Policies:", t.DBRPs)
```
Comment (jsotelo) [inline cmd/kapacitor/main.go]: Typo here, should be "Retention"

Comment (nathanielc) [inline cmd/kapacitor/main.go]: @jsotelo Thanks fixing

**Comment by absolutejam** [comment]
Should this functionality work through all nodes? 

I've tried this with `var measurement = "Percent_Processor_Time"` then 
```
`    |where(lambda: measurement > crit)
```
and it failed. Is this a limitation or have I fudged the syntax?

**Comment by nathanielc** [comment]
You probably want `var measurement = 'Percent_Processor_Time'`, single quotes are strings.

**Pull request merged**